### PR TITLE
#77 epic games free offers notifications

### PR DESCRIPTION
**Pull request by Latemus**
Now bot will notify every thursday at 18.05 about new free games in Epic Games Store if setting set on. In addition, user can initiate epic games fetch in the current chat with command '/epicgames'

Example result:
image

Commit-msg
- Now bob fetches and displayes current free game deals from Epic Games Store. To enable automatic announcements on every thursday at 18.05 the announcment setting should be toggled on in a chat (off by default). Manual free games check can be initiated with command '/epicgames'
- No state is yet saved so does not support daily deals. Can be extended to save state of announced games to database so that free games status could be checked daily and only announced if there would be new games

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,12 @@ kuuluttaa uusimman gitin commit viestin käynnistyessään.
 satunnaisesti 2 - n vaihtoehdon välillä, kun ne on eroteltu avainsanalla 
 **".vai"**
 - `/kunta` generoi satunnaisen kunnan
+- `/kysymys` - Päivän kysymykseen liittyvien toimintojen hallinta
+- `/epicgames` - hakee tiedon kysymyshetkellä epic games storessa ilmaiseksi jaossa olevista peleistä.
+
+Muita ominaisuuksi:
```
Comment (taturl) [inline README.md:55]:
```suggestion
Muita ominaisuuksia:
```

File: README.md [hunk 2]
```diff
@@ -49,6 +49,12 @@ kuuluttaa uusimman gitin commit viestin käynnistyessään.
...
+- botti ylläpitää "päivän kysymys"-peliä. Pelissä yksi käyttäjä esittää päivän aikana kysymyksen, johon muut ryhmäläiset vastaavat. Voittaja ilmoitetaan vapaamuotoisesti, jolloin voittanut käyttäjä voi esittää seuraavan päivänä seuraava päivän kysymyksen. Botti pitää kirjaa näistä kysymyksistä, vastauksista ja voitoista, jos chattiin on luotu päivän ksymyksen kausi. Pelin pääsee aloittamaan päivän kysymyksen valikon kautta komennolla '/kysymys'
```
Comment (taturl) [inline README.md:56]:
```suggestion
- Botti ylläpitää "päivän kysymys" -peliä. Pelissä yksi käyttäjä esittää päivän aikana kysymyksen, johon muut ryhmäläiset vastaavat. Voittaja ilmoitetaan vapaamuotoisesti, jolloin voittanut käyttäjä voi esittää seuraavana päivänä seuraavan päivän kysymyksen. Botti pitää kirjaa näistä kysymyksistä, vastauksista ja voitoista, jos chattiin on luotu päivän kysymyksen kausi. Pelin pääsee aloittamaan päivän kysymyksen valikon kautta komennolla '/kysymys'
```

File: bobweb/bob/utils_common.py [hunk 3]
```diff
@@ -76,6 +76,14 @@ def flatten(item: any) -> List:
     return item[:1] + flatten(item[1:])
 
 
+def flatten_single(list_to_flatten: any) -> List:
+    flat_list = []
+    for sublist in list_to_flatten:
+        for item in sublist:
+            flat_list.append(item)
+    return flat_list
```
Comment (taturl) [inline bobweb/bob/utils_common.py:84]:
```suggestion
def flatten_single(potential_list: any) -> List: 
    new_list = [] 
    for e in potential_list: 
        if isinstance(e, list): 
            new_list.extend(flatten(e)) 
        else: 
            new_list.append(e) 
        return new_list 
```

**Review by taturl — APPROVED** [review]
Cool!

File: bobweb/bob/utils_common.py (hunk 3 above)
Comment (Latemus) [reply bobweb/bob/utils_common.py:84]: Tuo flatten_single olikin lopulta turha, niin nappaan sen pois ja vaihdan sen kutsun kutsumaan tuota flatten-funktiota. Sillä pelkällä flattenilla on testitkin ok.

**Pull request merged**